### ISSUE TITLE
panic when closing live preview with a line edit from the properties panel focussed, and re-opening

### ISSUE THREAD

**Issue by mitchgrout** [slint-ui/slint#5698]
To reproduce, set the focus to a line edit in the property editor, and close the preview while the line edit has the focus.
Then re-open the preview (through the "show preview" label in the editor) and observe the panic

### original description

A bit of an odd one.

Starting with en empty component, open the live editor, and set the `width` field to `self.width`. However, do not press enter or in any other way accept this value. Close the editor.
At this point, the `.slint` file should still contain an empty component. 
Open the live editor. This triggers the following panic:

```
thread 'main' panicked at D:\a\slint\slint\target\x86_64-pc-windows-msvc\release\build\slint-lsp-8eb16ecdb39ea5da\out\main.rs:34729:189:
called `Option::unwrap()` on a `None` value
```

Triggered on VSCode Slint v1.7.1

**Comment by ogoffart** [comment]
It can be reproduced just by having the focus on a line edit in the property editor while closing the preview.  (updated the description)

**Comment by ogoffart** [comment]
I think the problem can be explained like so:
```
export component PropertyView {
    in-out property <[Property]> properties: [ { name: "Hello"} ];
    VerticalLayout {
        for property in properties: HorizontalLayout {
            if !property.is-simple: LineEdit {
            }
            Rectangle {}
        }
        Rectangle {}
    }
}
```

The LineEdit has the focus, but the windows is hidden.
Clicking to "show preview" will change the `properties` model. this makes everything dirty, but nothing is refreshed.
Then the window is shown. A focus_event is sent and Slint will try to get the geometry of the LineEdit for the IME.
First it will get the width: that depends on the parent's layout which depends on the parent layout which depends on the content on the layout which then will instantiate new component and destroy the old ones.
Then it will querry the height, which depends on the parent's layout, but the parent is already destroyed so the `upgrade()` will fail (#3464)
